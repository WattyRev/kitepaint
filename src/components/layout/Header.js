import React from "react";
import styled from "styled-components";
import UserContainer from "../../containers/UserContainer";
import { A, P, H2, Icon } from "../../theme";

export const StyleWrapper = styled.div`
  background: ${({ theme }) => theme.colors.silver};
  height: 48px;
  width: 100%;
  display: flex;
  align-items: center;
  padding: 0 8px;
  box-shadow: 0 0 1px ${({ theme }) => theme.colors.black};
  position: relative;
  box-sizing: border-box;

  .left,
  .right {
    display: flex;
    algin-items: center;
    flex-grow: 1;
  }
  .right {
    justify-content: flex-end;
  }
`;

/**
 * The header at the top of the application.
 */
const Header = () => (
  <StyleWrapper>
    <div className="left">
      <H2>
        <A href="#">KitePaint</A>
      </H2>
    </div>
    <div className="right">
      <UserContainer>
        {userData =>
          userData.props.isLoggedIn && (
            <P>
              <A onClick={userData.actions.logOut} className="testing_sign-out">
                Sign Out <Icon icon="sign-out-alt" />
              </A>
            </P>
          )
        }
      </UserContainer>
    </div>
  </StyleWrapper>
);

export default Header;
