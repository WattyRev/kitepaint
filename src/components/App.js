import React from "react";
import { Route, Switch } from "react-router-dom";
import { isEmbedded } from "../utils/embed";
import Header from "./layout/Header";
import Footer from "./layout/Footer";
import RestrictedRoute from "./RestrictedRoute";
import About from "./about";
import Account from "./account";
import Activate from "./activate";
import Create from "./create";
import CreateNew from "./createNew";
import Designs from "./designs";
import Edit from "./edit";
import ErrorPage from "./ErrorPage";
import Home from "./home";
import MyDesigns from "./my-designs";
import View from "./view";

/**
 * The entry for the app. The router will go here.
 */
const App = () => (
  <React.Fragment>
    {!isEmbedded && <Header />}
    <Switch>
      <RestrictedRoute exact path="/" component={Home} />
      <RestrictedRoute exact path="/about" component={About} />
      <RestrictedRoute exact path="/account" component={Account} />
      <RestrictedRoute
        exact
        path="/activate/:userId/:activationCode"
        component={Activate}
      />
      <RestrictedRoute exact path="/create" component={Create} />
      <Route exact path="/create/:productId" component={CreateNew} />
      <RestrictedRoute exact path="/designs" component={Designs} />
      <RestrictedRoute exact path="/edit/:designId" component={Edit} />
      <RestrictedRoute exact path="/my-designs" component={MyDesigns} />
      <Route exact path="/view/:designId" component={View} />
      <Route component={ErrorPage} />
    </Switch>
    <Footer />
  </React.Fragment>
);

export default App;
