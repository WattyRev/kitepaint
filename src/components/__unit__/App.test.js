import React from "react";
import { shallow } from "enzyme";
import Header from "../layout/Header";
import App from "../App";

it("renders without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("displays the header when not embedded", () => {
  expect.assertions(1);
  const wrapper = shallow(<App _isEmbedded={false} />);
  expect(wrapper.find(Header)).toHaveLength(1);
});

it("does not display the header if the app is embedded", () => {
  expect.assertions(1);
  const wrapper = shallow(<App _isEmbedded />);
  expect(wrapper.find(Header)).toHaveLength(0);
});
