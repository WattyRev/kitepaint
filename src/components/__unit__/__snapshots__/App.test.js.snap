// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/account"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/create"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/create/:productId"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/designs"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/edit/:designId"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/view/:designId"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/account"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/create/:productId"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/designs"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/edit/:designId"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/view/:designId"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/account",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/create/:productId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/designs",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/edit/:designId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/view/:designId",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/account"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/create/:productId"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/designs"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/edit/:designId"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/view/:designId"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/account"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/create"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/create/:productId"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/designs"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/edit/:designId"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/view/:designId"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/account",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/create/:productId",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/designs",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/edit/:designId",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/view/:designId",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
